backend/svirt.pm

```diff
@@ -167,25 +170,33 @@ sub save_snapshot {
 
 sub load_snapshot {
     my ($self, $args) = @_;
-    my $snapname = $args->{name};
-    my $vmname   = $self->console('svirt')->name;
-    my $rsp;
+    my $snapname                   = $args->{name};
+    my $vmname                     = $self->console('svirt')->name;
     my $post_load_snapshot_command = '';
+    my $vm_state;
+    my $rsp;
     if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
         my $ps = 'powershell -Command';
+        $self->run_ssh_cmd("$ps (Get-VMCheckpoint -VMName $vmname).Name");
+        (undef, $vm_state) = $self->run_ssh_cmd("$ps (Get-VM -VMName $vmname).State", wantarray => 1);
```

Comment:
General question. Does powershell provide usefull exit-codes, and maybe we should check these as well?

Reply:
It does, but does not reflect the VM state though. We can possibly write `Get-VM -ComputerName Server1 | Where-Object {$_.State -eq 'Running'}` but using PowerShell over ssh with pipes is a bit cumbersome.